extension/chrome/elements/pgp_block_modules/pgp-block-signature-module.ts

```diff
@@ -18,11 +19,12 @@ export class PgpBlockViewSignatureModule {
 
   public renderPgpSignatureCheckResult = (signature: VerifyRes | undefined) => {
     if (signature) {
-      const signerEmail = signature.contact ? signature.contact.name || this.view.senderEmail : this.view.senderEmail;
+      const signerEmail = signature.signer?.primaryUserId ? Str.parseEmail(signature.signer.primaryUserId).email || this.view.senderEmail : this.view.senderEmail;
```

Comment:
Should we remove the `this.view.senderEmail` here? And just stick to the signer, since sender can be spoofed.

Reply:
Then if it's not defined, use `(unknown signer)`